### PR TITLE
cerebral-debugger: init at 3.1.0

### PR DESCRIPTION
**Pull request by scalavision**
###### Motivation for this change

Cerebral is a javascript library that you might have to deal with if you are a front-end javascript developer. Making
it work in NixOS was not straight forward, since it uses `electron` under the hood, so it would be rather nice
having it already in nixpkgs.

###### Things done

- [x ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x ] Ensured that relevant documentation is up to date
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

some CAVEATS:

* The app.asar binary is used directly, hence this is not built from source
* The cerebral is upgraded to electron 4, this version still have security issues

I have notified the author of the library about the security issues, using version 4 of electron instead of version 1 seems not affecting the app. I have used it quite a lot without any problems.

I would be happy to build this completely from source, but would need some guidance, as I have no experience with electron apps and very little experience using node on nixpkgs.

### CODE REVIEW

File: pkgs/development/cerebral-debugger/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{ stdenv
+, fetchurl
+, makeWrapper
+, electron_4
+}:
+let
+  electron = electron_4;
```
Comment (SuperSandro2000) [inline pkgs/development/cerebral-debugger/default.nix:7]: Just use electron_4 directly.

File: pkgs/development/cerebral-debugger/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+in
+stdenv.mkDerivation rec {
+
+  pname = "cerebral-debugger";
+  version = "3.1.0";
+
+  src = fetchurl {
+    url = "https://github.com/cerebral/cerebral-debugger/releases/download/v3.1.0/cerebral-debugger-3.1.0.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/cerebral-debugger/default.nix:15]:
```suggestion
    url = "https://github.com/cerebral/cerebral-debugger/releases/download/v${version}/cerebral-debugger-${version}.tar.gz";
```

File: pkgs/development/cerebral-debugger/default.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      easy to trace all state changes over time, and to see the state value
+      at any particular time.
+    '';
+    homepage = "https://github.com/cerebral/cerebral-debugger";
+    license = licenses.mit;
+    maintainers = with maintainers; [ scalavision ];
+    platforms = platforms.linux;
+
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/cerebral-debugger/default.nix:57]:
```suggestion
    platforms = platforms.linux;
  };
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>cerebral-debugger</li>
  </ul>
</details>

**Comment by scalavision** [comment]
Thanks @SuperSandro2000 . I have fixed the things you commented I believe :-)

Hm.. I am not sure what is meant with the `1 package marked as broken and skipped` part. The problem is that electron 4 is broken due to security issues. To enable it, you have to do this in `~/.config/nixpkgs/config.nix` :

```nix
{
   permittedInsecurePackages = [
     "electron-4.2.12"
   ];
}
```

This is of course very unfortunate, but dealing with more or less legacy software often is ..

File: pkgs/development/cerebral-debugger/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, fetchurl
+, makeWrapper
+, electron_4
+}:
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/development/cerebral-debugger/default.nix:7]:
```suggestion
}:

stdenv.mkDerivation rec {
```

File: pkgs/development/cerebral-debugger/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      having cerebral-debugger available. The Cerebral Debugger makes it
+      easy to trace all state changes over time, and to see the state value
+      at any particular time.
+    '';
+    homepage = "https://github.com/cerebral/cerebral-debugger";
+    license = licenses.mit;
+    maintainers = with maintainers; [ scalavision ];
+    platforms = platforms.linux;
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/cerebral-debugger/default.nix:53]:
```suggestion
  };
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>cerebral-debugger</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> Hm.. I am not sure what is meant with the 1 package marked as broken and skipped part.

Probably the marked insecure part.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cerebral-debugger</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
I don't like the idea of adding something that relies on something marked insecure. I think NUR could be a place for this but for nixpkgs it is to old. Sorry.

**Pull request closed without merging**